Packs/CommonScripts/Scripts/IsDomainInternal/IsDomainInternal.py

```diff
@@ -0,0 +1,93 @@
+from time import sleep
+
+import demistomock as demisto  # noqa: F401
+from CommonServerPython import *  # noqa: F401
+
+POLLING_TIME = 2  # Interval to wait in seconds when polling to check if indicator was created
+
+
+def check_domain_internal(domain, internal_domains):
+    parts = domain.split(".")
+    for i in range(len(parts), 0, -1):
+        sub = ".".join(parts[-i:])
+        if sub in internal_domains:
+            return True
+    return False
+
+
+def create_markdown_table(domain_list):
+    domain_list = sorted(domain_list, key=lambda x: not x['Internal'])
+    markdown_string = "| Domain Name | Internal |\n| --- | --- |\n"
+    for domain in domain_list:
+        markdown_string += f"| {domain['Name']} | {domain['Internal']} |\n"
+    return markdown_string
+
+
+internal_domains_list = demisto.args().get("InternalDomainsListName", "InternalDomains")
+domains_to_check = argToList(demisto.args().get("Domains", None))
+
+# Get the list of internal domains from the XSOAR list:
+internal_domains = demisto.executeCommand("getList", {"listName": internal_domains_list})[0]['Contents']
```

Comment:
is there always such a list? won't we get a `[][0]`?

Reply:
What do you mean by [][0]?
The list should exist as it's shipped outside the box. If it doesn't exist, the [0]['Contents'] will still exist, it will just contain the message "Item not found" which should be caught in the if statement